nssp_demo/postprocess_scoring.R

```diff
@@ -0,0 +1,412 @@
+script_packages <- c(
+  "dplyr",
+  "scoringutils",
+  "lubridate",
+  "ggplot2",
+  "argparser"
+)
+
+## load in packages without messages
+purrr::walk(script_packages, \(pkg) {
+  suppressPackageStartupMessages(
+    library(pkg, character.only = TRUE)
+  )
+})
+
+
+epiweek_to_date <- function(epiweek, epiyear) {
```

Comment:
this is incorrect

``` r
library(lubridate)
#> 
#> Attaching package: 'lubridate'
#> The following objects are masked from 'package:base':
#> 
#>     date, intersect, setdiff, union
epiweek_to_date <- function(epiweek, epiyear) {
  # Create date for January 1st of the epiyear
  jan1 <- as.Date(paste0(epiyear, "-01-01"))
  # Calculate days to add (epiweeks start on Sunday)
  days_to_add <- (epiweek - 1) * 7
  # Add days and adjust to previous Sunday
  date <- jan1 + days_to_add
  date <- date - lubridate::wday(date, week_start = 7)
  return(date)
}

my_date <- ymd("1900-01-01")
epiweek_to_date(epiweek(my_date), epiyear(my_date))
#> [1] "1899-12-30"
```

<sup>Created on 2024-11-04 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>